src/Data/CodeGen.purs

```diff
@@ -146,16 +147,28 @@ writeCodeFromAbi opts abiPath destFile = do
     ejson <- jsonParser <$> liftAff (readTextFile UTF8 abiPath)
     json <- either (liftAff <<< throwError <<< error) pure ejson
     (Abi abiWithErrors) <- either (liftAff <<< throwError <<< error) pure $ parseAbi opts json
-    abi <- for abiWithErrors case _ of
+    abiUnAnn <- for abiWithErrors case _ of
       Left (AbiDecodeError err) -> do
         tell [ ABIError { abiPath, error: err.error, idx: err.idx } ]
         pure Nothing
-      Right res -> pure $ Just $ Identity res
-    genCode (Abi $ catMaybes abi) { exprPrefix: opts.exprPrefix, indentationLevel: 0 }
+      Right res -> pure $ Just $ Identity $ res
+    let abi = map Identity $ maybeAnnotateArity $ map (\(Identity a) -> a) $ catMaybes abiUnAnn
+    genCode (Abi $ abi) { exprPrefix: opts.exprPrefix, indentationLevel: 0 }
       # runImported opts destFile
       # writeTextFile UTF8 destFile
       # liftAff
 
+maybeAnnotateArity :: Array AbiType -> Array AbiType
+maybeAnnotateArity abi = evalState (traverse go abi) S.empty
+  where
+    go :: AbiType -> State (S.Set String) AbiType
+    go fun@(AbiFunction (SolidityFunction f)) = do
+      st <- get
+      if f.name `S.member` st
+         then pure $ AbiFunction $ SolidityFunction f {name = f.name <> show (length f.inputs)}
```

Comment:
In the compiled output we have `safeTransferFrom` and `safeTransferFrom4`. if in source code/abi `safeTransferFrom` with 3 arg and `safeTransferFrom` with 4 arg are reordered  then we would have 
`safeTransferFrom` and `safeTransferFrom3`. where `safeTransferFrom` takes 4 argument.


I would expect all functions with same name to have the arity number, and this changing order of functions will not change names of this functions.